src/cascadia/TerminalControl/SearchBoxControl.cpp

```diff
@@ -209,4 +224,143 @@ namespace winrt::Microsoft::Terminal::Control::implementation
+    // - Handler for clicking the case sensitivity toggle. Triggers SearchChanged event
+    // Arguments:
+    // - sender: not used
+    // - e: not used
+    // Return Value:
+    // - <none>
+    void SearchBoxControl::CaseSensitivityButtonClicked(winrt::Windows::Foundation::IInspectable const& /*sender*/, winrt::Windows::UI::Xaml::RoutedEventArgs const& /*e*/)
+    {
+        _SearchChangedHandlers(TextBox().Text(), _GoForward(), _CaseSensitive());
+    }
+
+    // Method Description:
+    // - Formats a status message representing the search state:
+    // * "Searching" - if totalMatches is negative
+    // * "No results" - if totalMatches is 0
+    // * "?/n" - if totalMatches=n matches and we didn't start the iteration over matches
+    // (usually we will get this after buffer update)
+    // * "m/n" - if we are currently at match m out of n.
+    // * "m/max+" - if n > max results to show
+    // * "?/max+" - if m > max results to show
+    // Arguments:
+    // - totalMatches - total number of matches (search results)
+    // - currentMatch - the index of the current match (0-based)
+    // Return Value:
+    // - status message
+    winrt::hstring SearchBoxControl::_FormatStatus(int32_t totalMatches, int32_t currentMatch)
+    {
+        if (totalMatches < 0)
+        {
+            return RS_(L"TermControl_Searching");
+        }
+
+        if (totalMatches == 0)
+        {
+            return RS_(L"TermControl_NoMatch");
+        }
+        std::wstringstream ss;
+
+        if (currentMatch < 0 || currentMatch > (MaximumTotalResultsToShowInStatus - 1))
+        {
+            ss << CurrentIndexTooHighStatus;
+        }
+        else
+        {
+            ss << currentMatch + 1;
+        }
+
+        ss << StatusDelimiter;
+
+        if (totalMatches > MaximumTotalResultsToShowInStatus)
+        {
+            ss << TotalResultsTooHighStatus;
+        }
+        else
+        {
+            ss << totalMatches;
+        }
+
+        return ss.str().data();
+    }
+
+    // Method Description:
+    // - Helper method to measure the width of the text block given the text and the font size
+    // Arguments:
+    // - text: the text to measure
+    // - fontSize: the size of the font to measure
+    // Return Value:
+    // - the size in pixels
+    double SearchBoxControl::_TextWidth(winrt::hstring text, double fontSize)
+    {
+        winrt::Windows::UI::Xaml::Controls::TextBlock t;
+        t.FontSize(fontSize);
+        t.Text(text);
+        t.Measure({ FLT_MAX, FLT_MAX });
+        return t.ActualWidth();
+    }
+
+    // Method Description:
+    // - This method tries to predict the maximal size of the status box
+    // by measuring different possible statuses
+    // Return Value:
+    // - the size in pixels
+    double SearchBoxControl::_GetStatusMaxWidth()
+    {
+        const auto fontSize = StatusBox().FontSize();
+        const auto maxLength = std::max({ _TextWidth(_FormatStatus(-1, -1), fontSize),

```

Comment:
Can we cache this value and only update it if we switch languages?

Reply:
Languages do not change at runtime. :)